### ISSUE TITLE
Default gems on Fedora are not loaded anymore

### ISSUE THREAD

**Issue by voxik** [rubygems/rubygems#5156]
Since I have pushed Ruby 3.0.3 into Fedora, there are reports such as [this](https://bugzilla.redhat.com/show_bug.cgi?id=2027099), that e.g. IRB does not work:

~~~
/usr/share/rubygems/rubygems/specification.rb:1402:in `rescue in block in activate_dependencies': Could not find 'reline' (>= 0.1.5) among 7 total gem(s) (Gem::MissingSpecError)
Checked in 'GEM_PATH=/root/.local/share/gem/ruby:/usr/share/gems:/usr/local/share/gems' at: /usr/share/gems/specifications/irb-1.3.5.gemspec, execute `gem env` for more information
	from /usr/share/rubygems/rubygems/specification.rb:1399:in `block in activate_dependencies'
	from /usr/share/rubygems/rubygems/specification.rb:1388:in `each'
	from /usr/share/rubygems/rubygems/specification.rb:1388:in `activate_dependencies'
	from /usr/share/rubygems/rubygems/specification.rb:1370:in `activate'
	from /usr/share/rubygems/rubygems.rb:299:in `block in activate_bin_path'
	from /usr/share/rubygems/rubygems.rb:298:in `synchronize'
	from /usr/share/rubygems/rubygems.rb:298:in `activate_bin_path'
	from /usr/bin/irb:23:in `<main>'
/usr/share/rubygems/rubygems/dependency.rb:311:in `to_specs': Could not find 'reline' (>= 0.1.5) among 7 total gem(s) (Gem::MissingSpecError)
Checked in 'GEM_PATH=/root/.local/share/gem/ruby:/usr/share/gems:/usr/local/share/gems' , execute `gem env` for more information
	from /usr/share/rubygems/rubygems/specification.rb:1400:in `block in activate_dependencies'
	from /usr/share/rubygems/rubygems/specification.rb:1388:in `each'
	from /usr/share/rubygems/rubygems/specification.rb:1388:in `activate_dependencies'
	from /usr/share/rubygems/rubygems/specification.rb:1370:in `activate'
	from /usr/share/rubygems/rubygems.rb:299:in `block in activate_bin_path'
	from /usr/share/rubygems/rubygems.rb:298:in `synchronize'
	from /usr/share/rubygems/rubygems.rb:298:in `activate_bin_path'
	from /usr/bin/irb:23:in `<main>'
~~~

From brief look. I'd blame #5044, because it postpones loading all the configurations, we do. And indeed, trying to move back the block of code loading the operating_system.rb, the default gems are listed again.

**Comment by voxik** [comment]
Ah, there is already #5154

**Comment by deivid-rodriguez** [comment]
Hello @voxik, sorry for the regression, I found out about it just yesterday, I imagined more stuff would be indeed broken, not only brew's formula. It will be fixed by #5154. I'll close this since the issue is already reported and fixed.
